storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogMetadataCache.java

```diff
@@ -302,22 +307,26 @@ public void addCopyInProgressSegment(RemoteLogSegmentMetadata remoteLogSegmentMe
 
         RemoteLogSegmentId remoteLogSegmentId = remoteLogSegmentMetadata.remoteLogSegmentId();
         RemoteLogSegmentMetadata existingMetadata = idToSegmentMetadata.get(remoteLogSegmentId);
-        checkStateTransition(existingMetadata != null ? existingMetadata.state() : null,
-                remoteLogSegmentMetadata.state());
-
+        boolean isValid = checkStateTransition(existingMetadata != null ? existingMetadata.state() : null,
+                remoteLogSegmentMetadata.state(), remoteLogSegmentMetadata.remoteLogSegmentId());
```

Comment:
Why don't we check `isInitialized` here?

Reply:
This is `addCopyInProgressSegment` block where it accepts only the `COPY_SEGMENT_STARTED` event (which is the first/starting event for a segment) so the `isInitialized` check is not required since all the other transition events should be available for that segmentId.